mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraSchemaVersionDAO.java

```diff
@@ -0,0 +1,67 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.mail;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor;
+import static org.apache.james.mailbox.cassandra.table.CassandraSchemaVersionTable.TABLE_NAME;
+import static org.apache.james.mailbox.cassandra.table.CassandraSchemaVersionTable.KEY;
+import static org.apache.james.mailbox.cassandra.table.CassandraSchemaVersionTable.VALUE;
+
+import java.util.concurrent.CompletableFuture;
+
+import static com.datastax.driver.core.querybuilder.QueryBuilder.eq;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.select;
+
+public class CassandraSchemaVersionDAO {
+
+    public static final int DEFAULT_VERSION = 1;
+    public static final int KEY_FOR_VERSION = 0;
+
+    private final PreparedStatement readVersionStatement;
+    private final CassandraAsyncExecutor cassandraAsyncExecutor;
+
+    public CassandraSchemaVersionDAO(Session session) {
+        cassandraAsyncExecutor = new CassandraAsyncExecutor(session);
+        readVersionStatement = prepareReadVersionStatement(session);
+    }
+
+    private static int versionFromRow(Row row) {
+        return row.getInt(VALUE);
+    }
+
+    private PreparedStatement prepareReadVersionStatement(Session session) {
+        return session.prepare(
+            select(VALUE)
+                .from(TABLE_NAME)
+                .where(eq(KEY, KEY_FOR_VERSION)));
+    }
+
+    public CompletableFuture<Integer> getCurrentSchemaVersion() {
```

Comment:
Here I would advocate having all the version related constants in the same place, so changing it is easier.

Hence I would simply return an optional here